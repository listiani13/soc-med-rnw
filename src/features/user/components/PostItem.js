import React, {Component} from 'react';
import {View, StyleSheet} from 'react-native';
import {Avatar, Button, TextField} from '@material-ui/core';
import {Text} from '../../../core-ui';
import {BORDER_RADIUS, BOX_SHADOW} from '../../../constants/layout';
import {LIGHT_GREY} from '../../../constants/color';
import CommentItem from './CommentItem';
import getInitialName from '../../../helpers/getInitialName';

type Comment = {
  postId: number,
  id: number,
  name: string,
  email: string,
  body: string,
};
type Props = {
  name: string,
  title: string,
  content: string,
  comments: Array<Comment>,
};
type State = {
  isCommentsShown: boolean,
};
export default class PostItem extends Component<Props, State> {
  state = {
    isCommentShown: false,
  };
  render() {
    let {name, title, content, comments} = this.props;
    let {isCommentShown} = this.state;

    return (
      <View style={styles.root}>
        <View style={styles.postHeader}>
          <View style={styles.identity}>
            <Avatar sizes="small" style={{marginRight: 10}}>
              {getInitialName(name)}
            </Avatar>
            <View style={styles.name}>
              <Text size="small">{name}</Text>
              <Text>{title}</Text>
            </View>
          </View>
          <Text size="xsmall" style={styles.post}>
            {content}
          </Text>
          <Button
            onClick={this._toggleComment}
            style={{alignSelf: 'flex-start'}}
          >
            {isCommentShown ? 'Hide' : 'Show'} Comments
          </Button>
        </View>
        {isCommentShown && (
          <View style={styles.commentList}>
            {comments.map((comment, index) => {
              let {name, body} = comment;
              return <CommentItem name={name} body={body} key={index} />;
            })}
            <TextField
              label="Add comment here"
              multiline
              rows="2"
              style={{marginBottom: 12}}
            />
            <Button
              variant="contained"
              color="primary"
              style={{alignSelf: 'flex-end', width: '25%'}}
            >
              Post
            </Button>
          </View>
        )}
      </View>
    );
  }
  _toggleComment = () => {
    this.setState({isCommentShown: !this.state.isCommentShown});
  };
}

const styles = StyleSheet.create({
  root: {
    borderRadius: BORDER_RADIUS,
    boxShadow: BOX_SHADOW,
    borderWidth: 1,
    borderColor: LIGHT_GREY,
    marginBottom: 15,
  },
  postHeader: {
    padding: 5,
  },
  identity: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  name: {
    marginBottom: 5,
  },
  post: {
    paddingVertical: 5,
  },
  commentList: {
    borderTopColor: LIGHT_GREY,
    borderTopWidth: 1,
    padding: 5,
    backgroundColor: LIGHT_GREY,
  },
});
