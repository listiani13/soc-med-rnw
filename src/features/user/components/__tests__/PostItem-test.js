// @flow
import React from 'react';
import {render} from 'react-testing-library';
import PostItem from '../PostItem';
import {COMMENTS} from '../../../../data/comment';

describe('Post Item component', () => {
  it('should render Post Item', () => {
    let {container} = render(
      <PostItem
        name="Name"
        title="Hello World"
        content="This is body"
        comments={COMMENTS}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
