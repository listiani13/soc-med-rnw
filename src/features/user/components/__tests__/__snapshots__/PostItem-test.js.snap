// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Item component should render Post Item 1`] = `
<div>
  <div
    class="rn-alignItems-1oszu61 rn-borderTopColor-478rc6 rn-borderRightColor-bmvd0i rn-borderBottomColor-y0r23w rn-borderLeftColor-12jqpwx rn-borderTopLeftRadius-kicko2 rn-borderTopRightRadius-notknq rn-borderBottomRightRadius-1dpl46z rn-borderBottomLeftRadius-6t5ypu rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-5kkj8d rn-borderRightWidth-13l2t4g rn-borderBottomWidth-qklmqi rn-borderLeftWidth-1ljd8xs rn-boxShadow-18cwage rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-ku1wi2 rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <div
      class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-glunga rn-paddingRight-19gegkz rn-paddingBottom-1inuy60 rn-paddingLeft-qhyqy2 rn-position-bnwqim rn-zIndex-1lgpqti"
    >
      <div
        class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-18u37iz rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
      >
        <div
          class="MuiAvatar-root-1 MuiAvatar-colorDefault-2"
          style="margin-right: 10px;"
        >
          N
        </div>
        <div
          class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-d0pm55 rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
        >
          <div
            class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
            dir="auto"
            style="font-size: 15px;"
          >
            Name
          </div>
          <div
            class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1i10wst rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
            dir="auto"
          >
            Hello World
          </div>
        </div>
      </div>
      <div
        class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-glunga rn-paddingRight-9aemit rn-paddingBottom-1inuy60 rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
        dir="auto"
        style="font-size: 13px;"
      >
        This is body
      </div>
      <button
        class="MuiButtonBase-root-30 MuiButton-root-4 MuiButton-text-6 MuiButton-flat-9"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label-5"
        >
          Show
           Comments
        </span>
        <span
          class="MuiTouchRipple-root-33"
        />
      </button>
    </div>
  </div>
</div>
`;
